File: src/ProjectR/content/scripts/spells/PoisonFlight.lua
function GetTargetType()
  return Enemies
end

function GetName()
  return "Poison Flight"
end

function GetDescription()
  return "Fly among the enemy ranks leaving a poison trail behind you.\nChance to inflict PAR and PSN."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .8
end

function GetDelay()
  return .25
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return { DRK }
end

function SpellEffect()
  local damage = (4 * aAD * (aDRK/100) - dDEF) * (100/dDRK)  
  d:ApplyDebuff(PAR, 70)
  d:ApplyDebuff(PSN, 70)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/PoisonFlight.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class PoisonFlight : SpellScriptBase
    {
        public override string Name { get { return "Poison Flight"; } }
        public override string Description { get { return "Fly among the enemy ranks leaving a poison trail behind you.\nChance to inflict PAR and PSN."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new[] { EleMastery.DRK, }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 80; } }
        public override double Delay { get { return .25; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (4 * cAD * (cDRK / 100) - tDEF) * (100 / tDRK);
            TryToApplyDebuff(DebuffResistance.PAR, 70);
            TryToApplyDebuff(DebuffResistance.PSN, 70);
            DealDamage(damage);
        }
    }
}